### PR TITLE
Integrating tooltip on Stacked Area

### PR DESCRIPTION
**Pull request by Golodhros**
Plugging tooltip to stacked area chart.

### CODE REVIEW

**Comment by Golodhros** [comment]
fullscreen_10_16_17__2_54_pm

File: src/charts/tooltip/TooltipComponent.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ import tooltip from './tooltipChart';
 export default class TooltipComponent extends React.Component {
 
     static PropTypes = {
+        children: PropTypes.node.isRequired,
```
Comment (jchen-eb) [inline src/charts/tooltip/TooltipComponent.js]: You can do this instead: https://reactjs.org/docs/typechecking-with-proptypes.html#requiring-single-child

Which also has the added benefit of making sure there's only one child.

File: src/charts/tooltip/TooltipComponent.js [hunk 2]
```diff
@@ -47,10 +70,44 @@ export default class TooltipComponent extends React.Component {
...
+        // Update Tooltip State
+        this.setState((state) => ({
+            ...state,
+            dataPoint,
+            topicColorMap,
+            x,
+            y,
+        }));
+
+        if (this.props.children.props.customMouseMove) {
```
Comment (vincent-eb) [inline src/charts/tooltip/TooltipComponent.js]: nit: For the three conditionals here could do destructuring so don't need to repeat all the nested-ness:
`let {children: {props: {customMouseMove}}} = this.props;`

**Comment by Golodhros** [comment]
Talked with Ben about this, we will refactor it into renderProps!

Merging this for now in order to unblock pending PRs, we will come back to it.

**Pull request merged**